joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstForExpressionsCreator.scala

```diff
@@ -365,6 +369,19 @@ trait AstForExpressionsCreator(implicit withSchemaValidation: ValidationMode) {
     astForSimpleCall(node.asSimpleCall)
   }
 
+  protected def astForYield(node: YieldExpr): Ast = {
+    scope.useProcParam match {
+      case Some(param) =>
+        astForMemberCall(
+          MemberCall(SimpleIdentifier()(node.span.spanStart(param)), ".", "call", node.arguments)(node.span)
```

Comment:
Could we lower this differently? My concern is that this is closer to reflection in how this is executed, and not sure that our call graph could resolve this.

Could this instead be a `SimpleCall`?

Reply:
Also, concerning flows, the yield needs to map to `MethodReturn`, how is this done in `pysrc2cpg`/should we add a `return` here?

Reply:
```scala
      case node: ast.Yield          => unhandled(node)
      case node: ast.YieldFrom      => unhandled(node)
```
it isn't handled there, but they would need to be handled vastly differently anyway.